conda_pypi/main.py

```diff
@@ -168,3 +194,138 @@ def ensure_target_env_has_externally_managed(command: str):
                 path.unlink()
     else:
         raise ValueError(f"command {command} not recognized.")
+
+
+def pypi_lines_for_explicit_lockfile(
+    prefix: Path | str, checksum: Literal["md5", "sha256"] | None = None
+) -> list[str]:
+    PrefixData._cache_.clear()
+    pd = PrefixData(str(prefix), pip_interop_enabled=True)
+    pd.load()
+    lines = []
+    python_record = list(pd.query("python"))
+    assert len(python_record) == 1
+    python_record = python_record[0]
+    python_details = {"version": ".".join(python_record.version.split(".")[:3])}
+    if "pypy" in python_record.build:
+        python_details["implementation"] = "pp"
+    else:
+        python_details["implementation"] = "cp"
+    for record in pd.iter_records():
+        if record.package_type != PackageType.VIRTUAL_PYTHON_WHEEL:
+            continue
+        ignore = False
+        wheel = {}
+        hashed_record = ""
+        for path in record.files:
+            path = Path(context.target_prefix, path)
+            if "__editable__" in path.stem:
+                ignore = True
+                break
+            if path.name == "direct_url.json" and path.parent.suffix == ".dist-info":
+                data = json.loads(path.read_text())
+                if data.get("dir_info", {}).get("editable"):
+                    ignore = True
+                    break
+            if checksum and path.name == "RECORD" and path.parent.suffix == ".dist-info":
+                hashed_record = compute_record_sum(path, checksum)
+            if path.name == "WHEEL" and path.parent.suffix == ".dist-info":
+                for line in path.read_text().splitlines():
+                    line = line.strip()
+                    if ":" not in line:
+                        continue
+                    key, value = line.split(":", 1)
+                    if key == "Tag":
+                        wheel.setdefault(key, []).append(value.strip())
+                    else:
+                        wheel[key] = value.strip()
+        if ignore:
+            continue
+        if record.url:
+            lines.append(f"# pypi: {record.url}")
+        else:
+            seen = {"abi": set(), "platform": set()}
+            lines.append(f"# pypi: {record.name}=={record.version}")
+            lines[-1] += f" --python-version {python_details['version']}"
+            lines[-1] += f" --implementation {python_details['implementation']}"
+            if wheel and (wheel_tag := wheel.get("Tag")):
+                for tag in wheel_tag:
+                    _, abi_tag, platform_tag = tag.split("-", 2)
+                    if abi_tag != "none" and abi_tag not in seen["abi"]:
+                        lines[-1] += f" --abi {abi_tag}"
+                        seen["abi"].add(abi_tag)
+                    if platform_tag != "any" and platform_tag not in seen["platform"]:
+                        lines[-1] += f" --platform {platform_tag}"
+                        seen["platform"].add(platform_tag)
```

Comment:
Oh no, manually parsing tag info :( This must be available in packaging somewhere

Reply:
True. Using `packaging.tags.parse_tag` now. Thanks!